Issue 953

intech opened the pull request:
add support multiple servers in transport nats: `nats://nats.server1:4222,nats://nats.server2:4222,nats://nats.server3:4222`

#953

icebob reviewed (changes requested):
Please add relevant tests

intech commented:
@icebob this fix works with current unit tests.
I made a quick fix to fix the issue for make a release.
From the tests, you can only add the indication of multiple servers, but now this is critical, it is rather an additional feature that still needs to be documented.

icebob commented:
I don't think because it should be failed this code: https://github.com/moleculerjs/moleculer/blob/master/test/unit/transporters/nats.spec.js#L392
And currently there is no use-case when the `url` contains multiple servers.

intech commented:
Yes, after your comment I decided to look through the tests and double-check that everything will be fine and saw this check on the url.
I corrected it and added one more test, but I do it via github ide in the browser and I may not get it the first time, sorry :)

icebob commented:
Nop, I'm waiting the CI and then I'll merge. Thanks, the quick job.

intech commented:
Ooook! Got to my computer and did a local test check. :)

icebob commented on test/unit/transporters/nats.spec.js:
```diff
@@ -389,13 +389,13 @@ describe("Tests Nats V2.x", () => {
 			let transporter = new NatsTransporter("nats://server1:4222,nats://server2:4222,nats://server3:4222");
 			transporter.isLibLegacy = jest.fn(() => false);
-			expect(transporter.opts).toEqual({ preserveBuffers: true, maxReconnectAttempts: -1, servers: ["server1:4222", "server2:4222", "server3:4222"] });
+			expect(transporter.opts).toEqual({ preserveBuffers: true, maxReconnectAttempts: -1, url: "nats://server1:4222,nats://server2:4222,nats://server3:4222" });
```
But the original issue was that it should be in `servers` and not `url`, wasn't it?

intech replied:
this cannot be done at the constructor level, since the nats version is checked later and therefore I had to transfer the formation of servers to the connection only on nats 2.x
https://github.com/moleculerjs/moleculer/blob/issue-953/src/transporters/nats.js#L147

icebob replied:
ok, you are right.
